controller/src/beerocks/master/db/db.h

```diff
@@ -564,6 +564,24 @@ class db {
     const std::list<sChannelScanResults> &get_channel_scan_results(const sMacAddr &mac,
                                                                    bool single_scan);
 
+    /**
+     * @brief Lock permission for channel scan
+     * 
+     * @param task_id: id of task requesting to lock the scan permission
+     * @return true on success
+     * @return false on failure (if failed to lock or locked by other task)
+     */
+    bool try_lock_scan_permission(int task_id);
```

Comment:
should be just `lock_scan_permission` and if it fails return false

Reply:
the `try` is to differ from `lock-until-resource-is-free`.
we try to take ownership and if we fail we return false.